src/framework/uicomponents/qml/MuseScore/UiComponents/ToggleButton.qml

```diff
@@ -0,0 +1,133 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+
+import MuseScore.Ui 1.0
+import MuseScore.UiComponents 1.0
+
+FocusScope {
+    id: root
+
+    property alias navigation: navCtrl
+
+    property bool checked: false
+
+    property string toolTipTitle: ""
+    property string toolTipDescription: ""
+    property string toolTipShortcut: ""
+
+    readonly property string checkedAccessibleString: root.checked ? qsTrc("ui", "checked", "checkstate")
+                                                                   : qsTrc("ui", "unchecked", "checkstate")
+
+    signal toggled()
+
+    implicitHeight: 20
+    implicitWidth: 36
+
+    opacity: root.enabled ? 1.0 : ui.theme.itemOpacityDisabled
+
+    function ensureActiveFocus() {
+        if (!root.activeFocus) {
+            root.forceActiveFocus()
+        }
+    }
+
+    NavigationControl {
+        id: navCtrl
+        name: root.objectName != "" ? root.objectName : "ToggleButton"
+        enabled: root.enabled && root.visible
+
+        accessible.role: MUAccessible.RadioButton
+        accessible.checked: root.checked
+
+        onTriggered: root.toggled()
+    }
+
+    Rectangle {
+        id: backgroundRect
+
+        anchors.fill: parent
+
+        color: root.checked ? ui.theme.accentColor : ui.theme.buttonColor
+
+        border.width: ui.theme.borderWidth
+        border.color: ui.theme.strokeColor
+        radius: 10
+
+        NavigationFocusBorder {
+            navigationCtrl: navCtrl
+        }
+
+        Rectangle {
+            id: handleRect
+
+            readonly property int margins: 2
+
+            anchors.verticalCenter: parent.verticalCenter
+            x: root.checked ? parent.width - width - margins : margins
```

Comment:
Would be nice to have the `x` property animated when the button is toggled. And it doesn't have hover/press states yet. But of course such refinements can be made later.

Reply:
Yes, but definitely later